.github/workflows/neuron-ci.yml

```diff
@@ -0,0 +1,243 @@
+              brew link openmpi
+              export CONFIG_OPTIONS="$CONFIG_OPTIONS --with-readline=no";
+            fi
+
+            export PATH=$INSTALL_DIR/x86_64/bin:$PATH;
+            ./.travis_build.sh $INSTALL_DIR;
+          fi
+
+          # basic tests for autoconf
+          if [ "$BUILD_MODE" == "autotools" ]; then
+            $PYTHON --version && $PYTHON -c 'import neuron; neuron.test()';
+          fi;
+
+          # basic test for cmake when python is not disabled
+          if [[ "$BUILD_MODE" == "cmake" && "$CMAKE_OPTION" != *"NRN_ENABLE_PYTHON=OFF"* ]]; then
+            $PYTHON --version && $PYTHON -c 'import neuron; neuron.test()';
+          fi;
+
+          # test neurondemo with cmake and autoconf
+          if [ "$BUILD_MODE" != "setuptools" ]; then
+            neurondemo -nogui -c 'demo(4)' -c 'run()' -c 'quit()';
+          fi;
+
+          # with cmake dynamic check python2 and python3 together
+          if [[ "$BUILD_MODE" == "cmake" && "$NRN_ENABLE_PYTHON_DYNAMIC" == "ON" ]]; then
+            $PYTHON3 -c 'import neuron; neuron.test()' && $PYTHON2 -c 'import neuron; neuron.test()';
+          fi;
+
+          # with autotools run rxd tests manually
+          if [ "$BUILD_MODE" == "autotools" ]; then
+            $PYTHON share/lib/python/neuron/rxdtests/run_all.py;
+          fi;
+
+          # with setuptools / pip, use wheel under virtual environment
+          if [ "$BUILD_MODE" == "setuptools" ]; then
+            neuron_wheel=wheelhouse/NEURON*.whl;
+            ./packaging/python/test_wheels.sh $PYTHON $neuron_wheel;
+          fi;
+          # with setuptools / pip, use wheel for global install
+          if [ "$BUILD_MODE" == "setuptools" ]; then
+            neuron_wheel=wheelhouse/NEURON*.whl;
+            ./packaging/python/test_wheels.sh $PYTHON $neuron_wheel false;
+          fi;
+        env:
+          BUILD_MODE: ${{ matrix.config.build_mode }}
+          NRN_ENABLE_PYTHON_DYNAMIC : ${{ matrix.config.python_dynamic }}
+          USE_PYTHON2 : ${{ matrix.config.use_python2 }}
+          INSTALL_DIR : ${{ runner.workspace }}/install
+      
+      # This step will set up an SSH connection on tmate.io for live debugging.
+      # To trigger it, simply prefix your branch name with `live-debug`
+      #   i.e.: live-debug-hocfix
+      - name: live debug session on failure
+        if: failure() && startsWith(github.ref, 'refs/heads/live-debug')
+        uses: mxschmitt/action-tmate@v3
+
+      - name: Documentation 
+        if: ${{ startsWith(matrix.os, 'ubuntu') && matrix.config.documentation == 'ON' }}
+        id: documentation
+        working-directory: ${{runner.workspace}}/nrn
+        run: |
+          echo "------- Build Doxygen Documentation -------";
+          pushd build;
+          make docs;
+          echo "-------- Disable jekyll --------";
+          pushd docs;
+          touch .nojekyll;
+          echo ::set-output name=status::done
+          
+#       - name: Deploy 🚀
```

Comment:
I've tried publishing to Github Pages,  but accessing GITHUB_TOKEN didn't work. @pramodk maybe you can have a look after we merge this, since you have repo rights.

Reply:
Ok. this can be done separately.